app/org/sagebionetworks/bridge/services/StudyService.java

```diff
@@ -154,6 +171,52 @@ public Study createStudy(Study study) {
 
         return study;
     }
+
+    public Study createSynapseProjectTeam(Long userId, Study study) throws SynapseException {
+        // create synapse project and team
+        Team team = new Team();
+        team.setName(randomAlphanumeric(5) + "Bridge-Team");
+        Project project = new Project();
+        project.setName(randomAlphanumeric(5) + "Bridge-Project");
+
+        Team newTeam = synapseClient.createTeam(team);
+
+        Project newProject = synapseClient.createEntity(project);
+
+        // modify project acl
+        org.sagebionetworks.repo.model.AccessControlList acl = synapseClient.getACL(newProject.getId());
+        // add exporter as admin
+        ResourceAccess toSet = new ResourceAccess();
+        toSet.setPrincipalId(Long.parseLong(EXPORTER_SYNAPSE_USER_ID));
+        toSet.setAccessType(ModelConstants.ENITY_ADMIN_ACCESS_PERMISSIONS);
+        acl.getResourceAccess().add(toSet);
+        // add user as admin
+        ResourceAccess toSetUser = new ResourceAccess();
+        toSetUser.setPrincipalId(userId); // passed by user as parameter
+        toSetUser.setAccessType(ModelConstants.ENITY_ADMIN_ACCESS_PERMISSIONS);
+        acl.getResourceAccess().add(toSetUser);
+
+        synapseClient.updateACL(acl);
+
+        // send invitation to target user for joining new team and grant admin permission to that user
+        MembershipInvtnSubmission teamMemberInvitation = new MembershipInvtnSubmission();
+        teamMemberInvitation.setInviteeId(userId.toString());
+        teamMemberInvitation.setTeamId(newTeam.getId());
+
+        synapseClient.createMembershipInvitation(teamMemberInvitation, null, null);
+        synapseClient.setTeamMemberPermissions(newTeam.getId(), userId.toString(), true);
+
+        String newTeamId = newTeam.getId();
+        String newProjectId = newProject.getId();
+
+        // finally, update study
+        study.setSynapseProjectId(newProjectId);
+        study.setSynapseDataAccessTeamId(Long.parseLong(newTeamId));
+        updateStudy(study, false);
+
+        return study;
```

Comment:
Yes, if there are already IDs for this project, I would throw an error. The user can delete these IDs if they want to redo their connection to Synapse, but I wouldn't overwrite this relationship as it may just break them.